dlt/common/storages/fsspec_filesystem.py

```diff
@@ -237,22 +281,29 @@ def glob_files(
     glob_result = fs_client.glob(filter_url, detail=True)
     if isinstance(glob_result, list):
         raise NotImplementedError(
-            "Cannot request details when using fsspec.glob. For ADSL (Azure) please use version"
+            "Cannot request details when using fsspec.glob. For adlfs (Azure) please use version"
             " 2023.9.0 or later"
         )
 
     for file, md in glob_result.items():
         if md["type"] != "file":
             continue
+        # filter out weird files that look like folders
+        size = int(md["size"])
+        if file.endswith("/") and size == 0:
+            continue
```

Comment:
I think this part may be confusing. There are no folders in buckets actually. And even if there is a strange object in a bucket, I think it still should be shown, if it fits the user's glob. Hiding them seems like unexpected behavior, because of which the user may stuck for some time.

Reply:
It's required to update the test files on `gs`. See this failure:

![image](https://github.com/dlt-hub/dlt/assets/26030071/cc38bec5-fc8f-4934-96f1-046643d84014)

It's only on `gs` bucket. But previously I saw the same failures on S3 and Azure, and I fixed it. There are two things needed:

1. In the file `tests/filesystem/samples/csv/freshman_lbs.csv` the headline must be like this:
```
"Sex","Weight(lbs,Sep)","Weight(lbs,Apr)","BMI(Sep)","BMI(Apr)"
```
2. Just delete this strange folder-like file that ends with `/`. On S3 I found it after you wrote, and deleted it, keeping only the `tests/filesystem/samples/gzip/taxi.csv.gz`, and it worked fine.

I did both of these things on S3 and Azure, and I would do on GS, but those credentials I have don't work, so I just don't have access to the GS bucket. However, this is not a functionality problem, there is just some mess in the test files on GS bucket.